api/osf_groups/views.py

```diff
@@ -0,0 +1,131 @@
+from api.osf_groups.permissions import IsGroupManager
+from api.osf_groups.serializers import OSFGroupSerializer, OSFGroupDetailSerializer
+from api.users.serializers import UserSerializer
+from framework.auth.oauth_scopes import CoreScopes
+from osf.models import OSFGroup
+
+
+class OSFGroupMixin(object):
+    """
+    Mixin with convenience method for retrieving the current OSF Group
+    """
+    group_lookup_url_kwarg = 'group_id'
+
+    def get_osf_group(self, check_object_permissions=True):
+
+        group = get_object_or_error(
+            OSFGroup,
+            self.kwargs[self.group_lookup_url_kwarg],
+            self.request,
+            display_name='osf_group',
+        )
+
+        if check_object_permissions:
+            self.check_object_permissions(self.request, group)
+        return group
+
+
+class OSFGroupList(JSONAPIBaseView, generics.ListCreateAPIView, ListFilterMixin, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON,)
+    serializer_class = OSFGroupSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-list'
+    ordering = ('-modified', )
+
+    def get_default_queryset(self):
+        return OSFGroup.objects.all()
+
+    # overrides ListCreateAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        """Create an OSFGroup.
+
+        :param serializer:
+        """
+        # On creation, logged in user is the creator
+        user = self.request.user
+        serializer.save(creator=user)
+
+
+class OSFGroupDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsGroupManager,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON,)
+    serializer_class = OSFGroupDetailSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-detail'
+    ordering = ('-modified', )
+
+    # Overrides RetrieveUpdateDestroyAPIView
+    def get_object(self):
+        return self.get_osf_group()
+
+    # Overrides RetrieveUpdateDestroyAPIView
+    def perform_destroy(self, instance):
+        auth = get_user_auth(self.request)
+        instance.remove_group(auth=auth)
```

Comment:
`remove_group` removes the django member group, manager group, and then deletes the OSF Group itself.   I thought this was cleaner than marking as deleted.

Deleting the django groups immediately removes any permissions the users had to particular nodes due to their osf group membership. Once the django groups were gone from the OSF Group, there was really no reason to keep the OSF Group itself (it has no members/managers).

Alternatively, we could mark the OSFGroup as deleted, and remove all the perms that its member django group has to all nodes.  API would have to change to prevent the deleted OSFGroup from being accessed. Open to feedback.